app/models/katello/presenters/sync_status.rb

```diff
@@ -0,0 +1,69 @@
+module Katello
+  module Presenters
+    class SyncStatus
+      include ::ActionView::Helpers::DateHelper
+      include ::Katello::TranslationHelper
+      STATUS_VALUES = {   :stopped      => _("Syncing Complete."),
+                          :error        => _("Sync Incomplete"),
+                          :never_synced => _("Never Synced"),
+                          :running      => _("Running"),
+                          :canceled     => _("Canceled"),
+                          :paused       => _("Paused")}.with_indifferent_access
+
+      def initialize(repo, task)
+        @repo = repo
+        @task = task
+      end
+
+      def sync_progress
+        return {:state => nil} unless @repo
+        return empty_task(@repo) unless @task
+        {   :id             => @repo.id,
+            :product_id     => @repo.product.id,
+            :progress       => {:progress => @task.progress * 100},
+            :sync_id        => @task.id,
+            :state          => format_state(@task),
+            :raw_state      => raw_state(@task),
+            :start_time     => format_date(@task.started_at),
+            :finish_time    => format_date(@task.ended_at),
+            :duration       => format_duration(@task.ended_at, @task.started_at),
+            :display_size   => @task.humanized[:output],
+            :size           => @task.humanized[:output],
+            :is_running     => @task.pending && @task.state != 'paused',
+            :error_details  => @task.errors
+        }
+      end
+
+      private
+
+      def empty_task(repo)
+        state = 'never_synced'
+        {   :id             => repo.id,
```

Comment:
Not sure about this format here with the blank space between `{` and the first key. Also, I'd prefer to see consistency when defining hashes like:

```
{
   :id =>
}
```